### PR TITLE
Add OSX support on travis-ci

### PR DESCRIPTION
**Pull request by jvoisin**
- Add OSX as a target in `.travis.yml`
- Remove the unsupported `-masm=intel` flag in the Makefile for OSX only
- Translate conditionally some Gnu-ism into OSX-ism in the asm implementation of sha1
- Fix some non portable `mktemp` invocation
- OSX has `shasum` instead of `sha1sum` or `sha1`

I also added some caching to travis.

### CODE REVIEW

File: test/test-airdecap-ng.sh [hunk 1]
```diff
@@ -30,7 +37,11 @@ cp -f "${TESTDIR}/wpa-psk-linksys.cap" "${tmpdir}"
         grep "Number of decrypted WPA  packets        53"
 [ $? -ne 0 ] && exit 1
 if ! type "sha1sum" > /dev/null 2>/dev/null ; then
+    if ! type "shasum" > /dev/null 2>/dev/null ; then
```
Comment (ZeroChaos-) [inline test/test-airdecap-ng.sh]: don't make running sha1sum dependant on a check for shasum.  we already verified we have sha1sum.  add this properly in the top level if/elif

File: test/test-airdecap-ng.sh [hunk 2]
```diff
@@ -45,10 +52,12 @@ cp -f "${TESTDIR}/wpa2-psk-linksys.cap" "${tmpdir}"
 ./airdecap-ng -e linksys -p dictionary "${tmpdir}/wpa2-psk-linksys.cap" | \
         grep "Number of decrypted WPA  packets        25"
 [ $? -ne 0 ] && exit 1
-if ! type "sha1sum" > /dev/null 2>/dev/null ; then
-        sha1sum=$(sha1 -q "${tmpdir}/wpa2-psk-linksys-dec.cap")
+if type "sha1sum" > /dev/null 2>/dev/null ; then
+    sha1sum=$(sha1sum "${tmpdir}/wpa2-psk-linksys-dec.cap" | awk '{print $1}')
+elif  type "shasum" > /dev/null 2>/dev/null ; then
+    sha1sum=$(shasum "${tmpdir}/wpa2-psk-linksys-dec.cap" | awk '{print $1}')
 else
```
Comment (ZeroChaos-) [inline test/test-airdecap-ng.sh]: else doesn't make sense here, fix it to use elif like that other stanza and else should likely just run "false"

**Comment by jvoisin** [comment]
I have rebased it on `master`

File: .travis.yml [hunk 3]
```diff
@@ -11,7 +20,25 @@ addons:
...
     - libnl-genl-3-dev
     - libpcre3-dev
+
 compiler:
   - gcc
   - clang
-script: make sqlite=true experimental=true && make check sqlite=true experimental=true && make clean && make sqlite=true experimental=true gcrypt=true && make check sqlite=true experimental=true gcrypt=true && make pcre=true && make check pcre=true && CFLAGS="-g -W -Wall -O0" make sqlite=true experimental=true && CFLAGS="-g -W -Wall -O0" make check sqlite=true experimental=true && make clean && CFLAGS="-g -W -Wall -O0" make sqlite=true experimental=true gcrypt=true && CFLAGS="-g -W -Wall -O0" make check sqlite=true experimental=true gcrypt=true && CFLAGS="-g -W -Wall -O0" make pcre=true && CFLAGS="-g -W -Wall -O0" make check pcre=true
+
+cache:
+    - ccache
```
Comment (ZeroChaos-) [inline .travis.yml]: don't use ccache

File: common.mak [hunk 4]
```diff
@@ -137,7 +137,7 @@ ifeq ($(OSNAME), OpenBSD)
 	CXXFLAGS	= $(CFLAGS) -fdata-sections -ffunction-sections
 else
 	INTEL_ASM	= $(shell echo | gcc -fsyntax-only -masm=intel -xc - 2>/dev/null && echo Y)
-ifeq ($INTEL_ASM), Y)
+ifeq ($(INTEL_ASM), Y)
```
Comment (ZeroChaos-) [inline common.mak:140]: please don't toss unrelated fixes into your PR

File: test/test-airdecap-ng.sh [hunk 5]
```diff
@@ -3,7 +3,21 @@
...
+compute_sha1() {
+    if type "sha1sum" > /dev/null 2>/dev/null ; then
+        sha1sum "${1}" | awk '{print $1}'
+    elif type "shasum" > /dev/null 2>/dev/null ; then
+        shasum "${1}" | awk '{print $1}'
+    elif type "sha1" > /dev/null 2>/dev/null ; then
+        sha1 -q "${1}"
+    else
+        echo "Unable to find something to compute sha1" 1>&2
+    fi
```
Comment (ZeroChaos-) [inline test/test-airdecap-ng.sh:17]: need an exit 1 here not just a nasty message otherwise this has to get back out of the function and it checks nothing against the string.

File: common.mak (hunk 4 above)
Comment (jvoisin) [inline common.mak:140]: I rebased against latest master, and eb6ed5776afc4f08ab8d0f31f02c471fcd145813 is breaking my PR because of a typo you made. What was I supposed to do?

File: .travis.yml (hunk 3 above)
Comment (jvoisin) [inline .travis.yml]: Why?
This will speed-up travis-ci.

Comment (ZeroChaos-) [inline .travis.yml]: https://wiki.gentoo.org/wiki/Handbook:X86/Working/Features#Caching_compilation_objects

ccache is widely known to cause random failures.  this isn't 2002, nor are we building chromium, it's not worth the risk of completely unpredictable failures.

**Comment by aircrack-ng** [comment]
We just committed a pretty big patch, could you update this PR?

**Comment by jvoisin** [comment]
This PR was already merged in 4182b5d .
